### ISSUE TITLE
Invalid behaviour of query action

### ISSUE THREAD

**Issue by rdulmina** [ballerina-platform/ballerina-lang#34916]
**Description:**

Consider the below sample

```bal

1. function testReachabilityWithQueryAction() returns string {
2.     string[] stringArray = ["Hello", " ", "World"];
3. 
4.     error? unionResult = from var item in stringArray
5.         where item is string
6.         do {
7.             if item == "Hello" {
8.                 return "Hello";
9.             }
10.         };
11. 
12.     if unionResult is error {
13.         return "ballerina";
14.     } else {
15.         return "c#";
16.     }
17. }
18. 
19. public function main() {
20.     io:println(testReachabilityWithQueryAction());
21. }

```

Prints `c#` where it should print `Hello`. Apparently the issue is when the return statement at L#8 is executed it returns to do block, not from the function

**Affected Versions:**
2201.0.0 (Swan Lake)

**Comment by pcnfernando** [comment]
Simplified version where the iteration does not brake with return statement
```ballerina
function abc() returns int? {
    error? res = from int i in 1 ... 3
        do {
            io:println(i);
            return 1;
        };
}

public function main() {
    _ = abc();
}


```

**Comment by lochana-chathura** [comment]
Fixed in #35344
